controllers/sms-gateway.js

```diff
@@ -119,6 +120,7 @@ module.exports = {
       .then(JSON.parse)
       .then(function(request) {
         // Process webapp-terminating messages asynchronously
+        // TODO these promises do nothing!
```

Comment:
They weren't making the code any more asynchronous and a loop would achieve the same result. This was a WIP commit and the TODO is now TODONE.